src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Optional;
+import mate.jdbc.exceptions.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String GET_BY_ID_QUERY = "SELECT * FROM manufacturers "
+            + "WHERE manufacturer_id = ? AND is_deleted = false";
+    private static final String GET_ALL_QUERY = "SELECT * FROM manufacturers "
+            + "WHERE is_deleted = false";
+    private static final String SOFT_DELETE_QUERY = "UPDATE manufacturers "
+            + "SET is_deleted = true WHERE manufacturer_id = ?";
+    private static final String UPDATE_QUERY = "UPDATE manufacturers SET manufacturer_name = ?,"
+            + "manufacturer_country = ? WHERE manufacturer_id = ? AND is_deleted = false";
+    private static final String INSERT_QUERY = "INSERT INTO manufacturers("
+            + "manufacturer_name, manufacturer_country) VALUES(?, ?);";
+    private static final String MANUFACTURER_ID_COLUMN = "manufacturer_id";
+    private static final String MANUFACTURER_NAME_COLUMN = "manufacturer_name";
+    private static final String MANUFACTURER_COUNTRY_COLUMN = "manufacturer_country";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        if (manufacturer == null) {
+            throw new RuntimeException("Manufacturer is null!");
+        }
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertManufactureStatement =
+                        connection.prepareStatement(INSERT_QUERY,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            insertManufactureStatement.setString(1, manufacturer.getName());
+            insertManufactureStatement.setString(2, manufacturer.getCountry());
+            insertManufactureStatement.executeUpdate();
+            ResultSet generatedKeys = insertManufactureStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                manufacturer.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't insert manufacture "
+                    + manufacturer + " to DB!", throwable);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(GET_BY_ID_QUERY)) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                manufacturer = createManufacturer(resultSet);
+            }
+        } catch (SQLException throwables) {
```

Comment:
check all methods please
```suggestion
        } catch (SQLException throwable) {
```

Reply:
done, sorry)